account-kit/signer/src/apiKeySigner.ts

```diff
@@ -0,0 +1,192 @@
+   *
+   * @param {ApiKeySignerClientParams} params The parameters for the client, including the API key and orgId
+   */
+  constructor(params: ApiKeySignerClientParams) {
+    const stamper = new ApiKeyStamper({
+      apiPublicKey: params.apiKey.publicKey,
+      apiPrivateKey: params.apiKey.privateKey,
+    });
+    this.turnkeyClient = new TurnkeyClient(
+      {
+        baseUrl: "https://api.turnkey.com",
+      },
+      stamper
+    );
+    this.alchemyClient = new AlchemySignerClient(params.connection);
+    this.orgId = params.orgId;
+  }
+
+  /**
+   * Retrieves the current user information.
+   *
+   * @returns {Promise<User>} A promise that resolves to the user object
+   */
+  public whoami = async (): Promise<User> => {
+    if (this.user) {
+      return this.user;
+    }
+
+    const stampedRequest = await this.turnkeyClient.stampGetWhoami({
+      organizationId: this.orgId,
+    });
+
+    const user = await this.alchemyClient.whoami(stampedRequest);
+    this.user = user;
+    return user;
+  };
+
+  /**
+   * Generates a stamped whoami request. Using this stamp is the most
+   * trusted way to get the user information since a stamp can only
+   * belong to the user who created it.
+   *
+   * @returns {Promise<TSignedRequest>} a promise that resolves to the "whoami" information for the logged in user
+   * @throws {Error} if no organization ID is provided
+   */
+  public stampWhoami = async (): Promise<TSignedRequest> => {
+    const user = await this.whoami();
+
+    return this.turnkeyClient.stampGetWhoami({
+      organizationId: user.orgId,
+    });
+  };
+
+  /**
+   * Signs a raw message
+   *
+   * @param {Hex} msg The message to be signed
+   * @param {SignRawMessageMode} mode The signing mode (default is "ETHEREUM")
+   * @returns {Promise<Hex>} A promise that resolves to the signed message
+   */
+  public signRawMessage = async (
+    msg: Hex,
+    mode: SignRawMessageMode = "ETHEREUM"
+  ): Promise<Hex> => {
+    const user = await this.whoami();
+
+    const stampedRequest = await this.turnkeyClient.stampSignRawPayload(
+      buildStampedSignatureRequestBody(user, msg, mode)
+    );
+
+    const { signature } = await this.alchemyClient.signPayload(stampedRequest);
+
+    return signature;
+  };
+}
+
+/**
+ * AlchemyApiKeySigner is a SmartAccountSigner that uses an ApiKeySignerClient to sign messages.
+ * Primarily intended to be used server-side.
+ *
+ * @example
+ * ```ts
+ *  const signer = new AlchemyApiKeySigner({
+ *   apiKey: {
+ *     privateKey: "private-api-key",
+ *     publicKey: "public-api-key",
+ *   },
+ *   connection: {
+ *     apiKey: "alchemy-api-key",
+ *   },
+ *   orgId: "user-org-id",
+ * });
+ *
+ * const account = await createModularAccountV2({
+ *   transport,
+ *   signer,
+ *   chain,
+ * });
+ *
+ * const client = createAlchemySmartAccountClient({
+ *   account,
+ *   transport,
+ *   chain,
+ * });
+ * ```
+ */
+export class AlchemyApiKeySigner implements SmartAccountSigner {
```

Comment:
Any reason for this to not just extend base alchemy signer? Again seems like quite a bit of duplication of code

Reply:
i def think that makes sense. i tried doing that initially but i believe i ran into some issues where we were trying to access local storage, which isn't available on the server. double checking though.

Reply:
actually `BaseAlchemySigner` is what requires localstorage, due to it creating a new `SessionManage` in the constructor, which fails in a non-browser environment.